CODING_GUIDLINES.md

```diff
@@ -0,0 +1,117 @@
+# Coding Guidelines and Recommendations
+
+The Stack:
+
+- Language: [TypeScript](https://www.typescriptlang.org/)
+- Framework: [React](https://reactjs.org/) + [Knockout](https://knockoutjs.com/)(Deprecated)
+- Bundler: [Webpack](https://webpack.js.org/)
+- Unit Test Runner: [Jest](https://jestjs.io/)
+- Code Formatter: [Prettier](https://prettier.io/)
+- Component Framework: [Fluent](https://developer.microsoft.com/en-us/fluentui#/)
+- Linter: [ESLint](https://eslint.org/) + [TSLint](https://palantir.github.io/tslint/)(Deprecated)
+- End to End Test Runner: [Jest](https://jestjs.io/)
+- End to End Test Framework: [Puppeteer](https://developers.google.com/web/tools/puppeteer)
+- Transpiler: [Babel](https://babeljs.io/) + [TypeScript](https://www.typescriptlang.org/)
+
+## Requirements
+
+Cosmos Explorer has been under constant development for over 5 years. As a result there are many different patterns and practices in the code base. This document serves as a guide to how we _should_ be writing code and helps avoid propagating practices which are no longer preferred. Each requirement in this document is labelled and color-coded to show the relative importance. In order from highest importance to lowest importance:
```

Comment:
```suggestion
Cosmos Explorer has been under constant development for over 5 years. As a result there are many different patterns and practices in the code base. This document serves as a guide to how we write code and helps avoid propagating practices which are no longer preferred. Each requirement in this document is labelled and color-coded to show the relative importance. In order from highest importance to lowest importance:
```

You should just assert that this is how we write code. `MUST` vs `SHOULD` is defined by your emojis below, so you don't need to add "_should_" here. Requirements are best done assertively.